File: medium/80.删除排序数组中的重复项-ii/80.删除排序数组中的重复项-ii.js
/*
 * @lc app=leetcode.cn id=80 lang=javascript
 *
 * [80] 删除排序数组中的重复项 II
 */
/**
 * @param {number[]} nums
 * @return {number}
 */
var removeDuplicates = function(nums) {
  let i = 0
  for (const n of nums) {
    if (i < 2 || n > nums[i - 2]) {
      nums[i++] = n
    }
  }
  return i
}

File: medium/80.删除排序数组中的重复项-ii/80.删除排序数组中的重复项-ii.py
#
# @lc app=leetcode.cn id=80 lang=python3
#
# [80] 删除排序数组中的重复项 II
#
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        i = 0
        for n in nums:
            if i < 2 or n > nums[i - 2]:
                nums[i] = n
                i += 1
        return i


